src/util/log.cc

```diff
@@ -553,8 +553,17 @@ std::ostream& operator<<(std::ostream& out, const std::exception_ptr& eptr) {
             } catch (...) {
                 out << ": " << std::current_exception();
             }
+        } catch (const std::nested_exception& ne) {
+            // if we got here, it means that the nested_exception is not mixed in
+            // with an object of type std::exception. It can happen in one of two
+            // cases:
+            // 1. The std::nested_exception was built and thrown using a normal
+            //    throw and not in the context of another catch with std::throw_with_nested
+            // 2. Somewhere in the code there is a throw of a type which doesn't derive from
+            //    std::exception and then it is rethrown using std::throw_with_nested.
+            out << ": " << "unknown nested type: " << ne.nested_ptr();
         } catch(...) {
-            // no extra info
+            out << " (" << "no extra info" << ")";
```

Comment:
Nitpick: this new "no extra info" message is unrelated to the issue of nested exceptions (you have an example in your test below), and I wonder if it's helpful... Basically the operator<< already printed the exception's type, but indeed doesn't know how to print any more information about its value. But I don't have strong feelings either way, so I think this is ok.

Reply:
I think this helps for completeness of the log message format. In the examples it can be seen the it makes the message more regular or uniform, using an example from https://github.com/scylladb/seastar/pull/875#discussion_r587290842 :
```
unknown_obj (no extra info)
std::runtime_error (Exception Level 0)
```
If anything, I would have added this `(no extra info)` also to the `unknown nested type` remark because it stands out.
On the other hand we should never see either of them if our exception throwing code is well written so I am not sure we
should devote a fix for that.
This is my opinion of course and it is totally subjective :slightly_smiling_face: